python/ray/experimental/serve/queues.py

```diff
@@ -220,3 +228,162 @@ def flush(self):
+        # perform traffic splitting for requests
+        for service, queue in self.queues.items():
+            # if there are incoming requests and there are backends
+            if len(queue) and len(self.traffic[service]):
+                while len(queue):
+                    # choose the next backend available from persistent
+                    # information
+                    chosen_backend = next(
+                        self.round_robin_iterator_map[service])
+                    request = queue.popleft()
+                    self.buffer_queues[chosen_backend].add(request)
+
+
+@ray.remote
+class RoundRobinPolicyQueueActor(RoundRobinPolicyQueue,
+                                 CentralizedQueuesActor):
+    pass
+
+
+class PowerOfTwoPolicyQueue(CentralizedQueues):
+    """
+    A wrapper class for powerOfTwo policy.
+    This backend selection policy is `Stateless` meaning
+    the current decisions of selecting backend are
+    dependent on previous decisions.
+    """
+
+    def _flush_service_queue(self):
+        # perform traffic splitting for requests
+        for service, queue in self.queues.items():
+            # while there are incoming requests and there are backends
+            while len(queue) and len(self.traffic[service]):
+                backend_names = list(self.traffic[service].keys())
+                backend_weights = list(self.traffic[service].values())
+                if len(self.traffic[service]) >= 2:
+                    # randomly pick 2 backends
+                    backend1, backend2 = np.random.choice(
+                        backend_names, 2, p=backend_weights)
+
+                    # see the length of buffer queues of the two backends
+                    # and pick the one which has less no. of queries
+                    # in the buffer
+                    if (len(self.buffer_queues[backend1]) <= len(
+                            self.buffer_queues[backend2])):
+                        chosen_backend = backend1
+                    else:
+                        chosen_backend = backend2
+                else:
+                    chosen_backend = np.random.choice(
+                        backend_names, p=backend_weights).squeeze()
+                request = queue.popleft()
+                self.buffer_queues[chosen_backend].add(request)
+
+
+@ray.remote
+class PowerOfTwoPolicyQueueActor(PowerOfTwoPolicyQueue,
+                                 CentralizedQueuesActor):
+    pass
+
+
+class FixedPackingPolicyQueue(CentralizedQueues):
+    """
+    A wrapper class for FixedPacking policy.
+    This backend selection policy is `Stateful` meaning
+    the current decisions of selecting backend are
+    dependent on previous decisions.
+    FixedPackingPolicy is k RoundRobin policy where first packing_num
+    queries are handled by 'backend-1' and next k queries are handled by
+    'backend-2' and so on ... where 'backend-1' and 'backend-2' are served
+    by the same service.
+
```

Comment:
Does the backend weights matter? Can we use it somehow or it is in direct conflict with the fixed packing solution

Reply:
We can have weighted versions of every policy! But I think they should be added as a separate policy

Reply:
i see separate policy make sense